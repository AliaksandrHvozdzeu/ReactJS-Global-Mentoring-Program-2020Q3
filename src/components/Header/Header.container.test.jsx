import React from "react";
import rerender from "react-test-renderer";
import HeaderContainer from "./js/Header.container";
import { jest } from "@jest/globals";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

describe("when AddButton", () => {

  const details = {
    "id": 337167,
    "title": "Fifty Shades Freed",
    "tagline": "Don't miss the climax",
    "vote_average": 6.1,
    "vote_count": 1195,
    "release_date": "2018-02-07",
    "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
    "budget": 55000000,
    "revenue": 136906000,
    "genres": [
      "Drama",
      "Romance"
    ],
    "runtime": 106
  };
  const preview = "DETAILS";
  const blur = true;
  const onLoadMovieById = jest.fn();
  const movieId = 19;
  const initialStore = {
    movies: {
      "genres": [
        "Drama",
        "Romance"
      ]
    }
  };
  const store = configureStore([])(initialStore);
  store.dispatch = jest.fn();

  test("then snapshot created", () => {
    const component = rerender.create(<Provider store={store}>
      <HeaderContainer preview={preview}
                       details={details}
                       blur={blur}
                       onLoadMovieById={onLoadMovieById}
                       movieId={movieId}
                       common={[]} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("then snapshot created when details is null", () => {
    const component = rerender.create(<Provider store={store}>
      <HeaderContainer preview={preview}
                       details={null}
                       blur={blur}
                       onLoadMovieById={onLoadMovieById}
                       movieId={movieId}
                       common={[]} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("then snapshot created when preview is null", () => {
    const component = rerender.create(<Provider store={store}>
      <HeaderContainer preview={null}
                       details={details}
                       blur={blur}
                       onLoadMovieById={onLoadMovieById}
                       movieId={movieId}
                       common={[]} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("then snapshot created when preview and details are null", () => {
    const component = rerender.create(<Provider store={store}>
      <HeaderContainer preview={null}
                       details={null}
                       blur={blur}
                       onLoadMovieById={onLoadMovieById}
                       movieId={movieId}
                       common={[]} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
