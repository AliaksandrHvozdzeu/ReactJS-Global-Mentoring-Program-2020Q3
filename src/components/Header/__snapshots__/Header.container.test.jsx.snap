// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when AddButton then snapshot created 1`] = `
<header
  className="blurred"
  id="header"
>
  <div
    className="preview-wrapper"
  >
    <div
      className="brand"
    >
      <p
        className="brand-title"
      >
        <a
          className="brand-title-name"
          href=" #"
          id="blink"
          onClick={[Function]}
        >
          netflix
        </a>
        roulette
      </p>
    </div>
    <div
      className="movie-preview unselectable"
    >
      <img
        alt="Movie poster"
        className="poster"
        onClick={[Function]}
        onError={[Function]}
        role="presentation"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
      <div
        className="movie-preview-description-block"
      >
        <h1
          className="movie-preview-title"
        >
          Fifty Shades Freed
        </h1>
        <p
          className="movie-preview-rating"
        >
          6.1
        </p>
        <p
          className="movie-preview-genres"
        />
        <p
          className="movie-preview-release"
        >
          2018
        </p>
        <p
          className="movie-preview-runtime"
        >
          106
        </p>
        <p
          className="movie-preview-description"
        >
          Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
        </p>
      </div>
    </div>
    <button
      className="close-preview-button"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <img
        alt="close preview"
        className="close-preview"
        src="../../../../public/images/search_42.png"
      />
    </button>
  </div>
</header>
`;

exports[`when AddButton then snapshot created when details is null 1`] = `
<header
  className="blurred"
  id="header"
>
  <div
    className="wrapper unselectable"
  >
    <div
      className="brand"
    >
      <p
        className="brand-title"
      >
        <a
          className="brand-title-name"
          href=" #"
          id="blink"
          onClick={[Function]}
        >
          netflix
        </a>
        roulette
      </p>
    </div>
    <div
      className="search"
    >
      <label
        className="search-title"
        htmlFor="search-field"
      >
        FIND YOUR MOVIE
        <div
          className="search-panel"
        >
          <input
            className="search-field"
            id="search-field"
            onChange={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            placeholder="What do you want to watch?"
            type="search"
            value=""
          />
          <button
            className="search-button"
            onClick={[Function]}
            type="button"
          >
            Search
          </button>
          <button
            className="add-button"
            onClick={[Function]}
            type="button"
          >
            + ADD MOVIE
          </button>
        </div>
      </label>
    </div>
  </div>
</header>
`;

exports[`when AddButton then snapshot created when preview and details are null 1`] = `
<header
  className="blurred"
  id="header"
>
  <div
    className="wrapper unselectable"
  >
    <div
      className="brand"
    >
      <p
        className="brand-title"
      >
        <a
          className="brand-title-name"
          href=" #"
          id="blink"
          onClick={[Function]}
        >
          netflix
        </a>
        roulette
      </p>
    </div>
    <div
      className="search"
    >
      <label
        className="search-title"
        htmlFor="search-field"
      >
        FIND YOUR MOVIE
        <div
          className="search-panel"
        >
          <input
            className="search-field"
            id="search-field"
            onChange={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            placeholder="What do you want to watch?"
            type="search"
            value=""
          />
          <button
            className="search-button"
            onClick={[Function]}
            type="button"
          >
            Search
          </button>
          <button
            className="add-button"
            onClick={[Function]}
            type="button"
          >
            + ADD MOVIE
          </button>
        </div>
      </label>
    </div>
  </div>
</header>
`;

exports[`when AddButton then snapshot created when preview is null 1`] = `
<header
  className="blurred"
  id="header"
>
  <div
    className="preview-wrapper"
  >
    <div
      className="brand"
    >
      <p
        className="brand-title"
      >
        <a
          className="brand-title-name"
          href=" #"
          id="blink"
          onClick={[Function]}
        >
          netflix
        </a>
        roulette
      </p>
    </div>
    <div
      className="movie-preview unselectable"
    >
      <img
        alt="Movie poster"
        className="poster"
        onClick={[Function]}
        onError={[Function]}
        role="presentation"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
      <div
        className="movie-preview-description-block"
      >
        <h1
          className="movie-preview-title"
        >
          Fifty Shades Freed
        </h1>
        <p
          className="movie-preview-rating"
        >
          6.1
        </p>
        <p
          className="movie-preview-genres"
        />
        <p
          className="movie-preview-release"
        >
          2018
        </p>
        <p
          className="movie-preview-runtime"
        >
          106
        </p>
        <p
          className="movie-preview-description"
        >
          Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
        </p>
      </div>
    </div>
    <button
      className="close-preview-button"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <img
        alt="close preview"
        className="close-preview"
        src="../../../../public/images/search_42.png"
      />
    </button>
  </div>
</header>
`;
